### DISCUSSION TITLE
restart the primary pod in supervised mode does not work

### DISCUSSION THREAD

**Discussion by ns-jhu** [cloudnative-pg/cloudnative-pg#1463] [General]
I created a pg cluster with 3 pods, but run in supervised mode 

I tried to test a cluster with 3 nodes running in supervised mode like this:

  primaryUpdateStrategy: supervised
  primaryUpdateMethod: restart

then I tried to test and see if the upgrade the image can trigger all pods to restart, turns out the 2 replica did restart and picked up the new image, but not the primary.
 even if I explicitly use cnpg plugin to restart the primary, it doesn't. I also tried to put the cluster in maintenance mode and restart, it does't restart either.

this is the command I run:
kubectl cnpg restart [cluster] [current_primary]

according to 
https://cloudnative-pg.io/documentation/current/rolling_update/#manual-updates-supervised
this should have worked. 
I even tried to put the supervised primary in maintenance mode first, that doesn't help at all.

the only way to make the primary restart is to demote it to a replica, restart it, then promote it back to primary.

I had tested in unsupervised mode before, all 3 nodes worked upgraded successfully .

either the document is wrong, or there is a bug.

**Comment by litaocdl** [comment]
@ns-jhu the cnpg command `kubectl cnpg restart [cluster] [current_primary]`
will have in-place restart if the target pod is primary, which will not have the pg image change. can you try to use `kubectl cnpg promote [cluster] [pod]` to trigger switch over to make the primary pg image upgraded.

**Reply by ns-jhu** [reply]
yes, that is the work around I am employing to get the primary upgraded. however, the document says

kubectl cnpg restart [cluster] [current_primary]

will start the primary, and I tested, it doesn't when the cluster is in supervised mode.  so either document is wrong or there is a bug. 

I tend to think this is a bug, as clearly the behavior doesn't match the document and the intention of the command.

**Reply by phisco** [reply]
It's behaving as intended, in-place updates are not possible when the restart would require a recreation of the whole instance. We should update the docs:

> This phase can only be completed with either a manual switchover or an in-place restart.

Should be something like:

> This phase can only be completed with either a manual switchover or an in-place restart. Keep in mind that image upgrades can not be applied by restarting in-place the instance, so a switchover is required in such cases.
